pkg/internal/client/txn.go

```diff
@@ -706,6 +707,11 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	// If we're not yet writing in this txn, but intend to, insert a
 	// begin transaction request before the first write command.
 	if needBeginTxn {
+		// If we're setting the system-config trigger, anchor the transaction to
```

Comment:
I don't expect that knowledge of system config details should leak into this high level client code. I'd be much happier with `Txn.SetTxnKey`.